### ISSUE TITLE
Finally completely broken by Fx59?

### ISSUE THREAD

**Issue by mastacheata** [Quicksaver/Tab-Groups#562]
Previously by enabling legacy extensions you could still use TabGroups from the keyboard and right-click shortcuts. (The management UI for creating/deleting groups and moving tabs between groups was broken ever since Firefox 57 betas) 

As of the latest Firefox 59 NIghtly builds from this week (wednesday or thursday, can't tell exactly) they broke the thing completely as you can't switch between groups anymore.
The bugs for introducing a tab-hiding API for WebExtensions haven't seen much work ever since 57 launched.

Has anyone found good alternatives?
The Vivaldi browser has a similar feature included, but instead of stacking and hiding the groups, they decided to stack and hide the tabs instead, displaying the groups in the tabbar instead of the tabs for each group.

**Comment by dgutov** [comment]
> The bugs for introducing a tab-hiding API for WebExtensions haven't seen much work

It landed today, actually.

**Comment by mastacheata** [comment]
Hahaha, I only saw the last post a few days ago, that they dropped discussions on wether or not to discard hidden tabs and would work on introducing the hiding feature first.

So that might give us hopes for a new TabGroup Extension for Firefox 59. :)
Too bad they already made my nice workaround of enabling legacy extensions break :(

**Comment by borsTiHD** [comment]
Try Sync Tab Groups.
I'm still on FF 56.0.2 and I followed the developer for a while, because I'm not wanting to give up on such an awesome feature (Tab Groups). My hole "internet" life depends on that addon.

But today I wanted to give it a try, update to FF 57 and install Sync Tab Groups.
Maybe its the best thing out there right now? Only thing I don't like is that all tabs on a "group switch" are gonna be loading completly. The developer said mozzila brings an update in the future, he could use to prevent the loading of all tabs.

https://addons.mozilla.org/en-US/firefox/addon/sync-tab-groups/

**Comment by dgutov** [comment]
You can also try (EDIT:) https://addons.mozilla.org/ru/firefox/addon/simple-tab-groups/. They have added a feature recently that mitigates that exact problem.

**Comment by BenjamenMeyer** [comment]
Tried Vivaldi a while back, but I can't stand their tab management; so waiting for FF to fix their stuff so either TabGroups can be ported or a similar proper extension can be done...in the mean time, sticking with FFv56 and using Chrome more too (for security)...though I really really miss Firefox+TabGroups.

**Comment by eridal** [comment]
so those like we still on v56 waiting for this terrific addon; what's the suggestion? wait for Firefox to enable the needed features? jump ships to a different addon?

I love this one, and would really like to stay in here. What are you guys planning/doing?

**Comment by irvin** [comment]
@eridal recently "Tab Hiding" API had been landed, and I assume there will be many similar add-ons appear after 59 soon. 

In the meantime, you can use Firefox Dev. 57 (57.0.0, no later version) with Tab group with following method, https://medium.com/@irvinfly/tab-groups-firefox-57-quantum-1d18b31e2dcd

**Comment by mastacheata** [comment]
The legacy.enable feature is still available, but the Tab-Groups extension apparently only works up to and including the very first alpha of Fx59.
Alpha2 broke this for me.

**Comment by scheinercc** [comment]
Sorry for this being more of a support request, but does one of you know a way to get the tabs/URLs of the hidden groups besides downgrading Nightly to a version that still works? I poked around in my profile, but couldn't find anything.

**Comment by GithubUser5462** [comment]
Try using this: https://addons.mozilla.org/en-US/firefox/addon/tab-center-redux/
All of your tabs should show up in the sidebar.

**Comment by scheinercc** [comment]
Thanks @GithubUser5462. That did the trick.

**Comment by Richie765** [comment]
Anyone knows something about this fork? https://github.com/amccausl/Tab-Groups
Looks like it may be promising but very little information and no install instructions.

**Comment by ebencollins** [comment]
@Richie765 Not sure about that specific extension, but I have been using [this one](https://github.com/nyordanov/panorama). It seems to be trying to replicate panorama. It's currently pretty limited but does at least allow me to have tab groups with tab hiding.
Addon page: https://addons.mozilla.org/en-US/firefox/addon/basic-panorama/.

**Comment by tofi86** [comment]
> Anyone knows something about this fork? https://github.com/amccausl/Tab-Groups
> Looks like it may be promising but very little information and no install instructions.

@amccausl can you elaborate on this? thanks! 👍

**Comment by mastacheata** [comment]
The extension sounds more promising than it looks.
The author merely made a sidebar for managing tabs into separate groups, but there's no sign of hiding any tabs on the original tabbar.
He even made it so that one tab can be in more than one group at a time if you open a new tab while having focused  a tab from group a while focusing group b in the sidebar. (This appears to be a bug more than a feature)

As far as building instructions go, take a look at my fork: https://github.com/mastacheata/Tab-Groups
I only did the bare neccessary modifications to make the extension build successfully and included a build script for windows.

You'll need the current NodeJS-LTS on your computer, then you can simply run npm run build-windows from the repo's root directory. 
That leads to the extension being built into the dist subdirectory.
In order to run the extension, you can either execute this command:
`node node_modules\.bin\web-ext --source-dir dist run --verbose`
Optionally include the --firefox="C:\Program Files\Mozilla Firefox Developer Edition\firefox.exe" switch to designate the path to firefox.exe which should be used (adjust path as neccessary).

Or you can skip that commandline stuff and simply load the dist/manifest.json from about:debugging as a temporary addon.
Note: Firefox Dev Edition or Nightly is required to load temporary addons as far as I'm informed. The stable version of Firefox won't load unsigned addons.

**Comment by amccausl** [comment]
Yeah, it's not ready for public consumption, but I added an install doc anyway.

Extension leaves the tab bar as is until hiding is available, can be hidden in browser CSS if desired.  Yes, that's a bug.  I'm working through a list of them before I'm comfortable showing it to people.

Thanks for the windows build script, will pull it :)

**Comment by mastacheata** [comment]
Hiding the whole tabbar isn't possible yet, but at least hiding single tabs should have landed recently.
Sent you a PR. Just made that fork to try your extension myself, but if you'd like to have it, feel free to use it. (Consider it public domain, I doubt it scratches the barrier for copyrightable content anyways)

**Comment by amccausl** [comment]
It's not possible programatically, but there's a manual workaround:

https://superuser.com/questions/1261660/firefox-quantum-ver-57-how-can-i-hide-the-horizontal-tab-bar-with-treesty/1261661

I didn't see the update for hiding single tabs, will take a look :)

**Comment by hdd60311** [comment]
@amccausl
support for hiding single tabs is here
https://bugzilla.mozilla.org/show_bug.cgi?id=1423725
https://developer.mozilla.org/en-US/Add-ons/WebExtensions/API/tabs/hide
